### PR TITLE
Add Core property to switch from `mmap` to `read` in IR Frontend

### PR DESCRIPTION
**Pull request by vurusovs**
### Details:
 - Continuation of https://github.com/openvinotoolkit/openvino/pull/12673

### Tickets:
 - 68937

### CODE REVIEW

File: src/inference/src/dev/core_impl.cpp [hunk 1]
```diff
@@ -1298,6 +1298,13 @@ void ov::CoreImpl::CoreConfig::set_and_update(ov::AnyMap& config) {
         _flag_allow_auto_batching = flag;
         config.erase(it);
     }
+
+    it = config.find(ov::ir_frontend_use_map_allocator.name());
+    if (it != config.end()) {
+        auto flag = it->second.as<bool>();
+        _flag_ir_fe_use_map_allocator = flag;
+        config.erase(it);
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.cpp:1306]: can we also update `get_property` to return current value of `enable_mmap` ?

File: src/frontends/ir/tests/frontend_test_basic.cpp [hunk 2]
```diff
@@ -343,6 +354,9 @@ TEST_F(IRFrontendTests, model_with_weights_reading_from_disk) {
     EXPECT_TRUE(res.valid) << res.message;
 }
 
+INSTANTIATE_TEST_SUITE_P(EnableMMapPropery, IRFrontendMMapTests, 
+                                ::testing::Values(true, false));
```
Comment (ilya-lavrenov) [inline src/frontends/ir/tests/frontend_test_basic.cpp:358]:
```suggestion
                                ::testing::Bool());
```

File: src/plugins/intel_cpu/src/exec_network.cpp [hunk 3]
```diff
@@ -114,7 +114,7 @@ ExecNetwork::ExecNetwork(const InferenceEngine::CNNNetwork &network,
         _callbackExecutor = std::make_shared<ImmediateSerialExecutor>();
 #else
         _callbackExecutor = _plugin->executorManager()->getIdleCPUStreamsExecutor(
-                                IStreamsExecutor::Config{"CPUCallbackExecutor", 1, 0, IStreamsExecutor::ThreadBindingType::NONE});
+                                IStreamsExecutor::Config{"CPUCallbacExecutor", 1, 0, IStreamsExecutor::ThreadBindingType::NONE});
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/exec_network.cpp:117]: revert?

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp [hunk 4]
```diff
@@ -679,6 +679,18 @@ TEST(OVClassBasicTest, SetTBBForceTerminatePropertyCoreNoThrow) {
     EXPECT_EQ(value, true);
 }
 
+TEST(OVClassBasicTest, SetEnableMmapPropertyCoreNoThrow) {
+    ov::Core ie;
+
+    bool value = true;
+    OV_ASSERT_NO_THROW(ie.set_property(ov::enable_mmap(false)));
+    OV_ASSERT_NO_THROW(value = ie.get_property(ov::enable_mmap.name()).as<bool>());
+    EXPECT_EQ(value, false);
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:688]:
```suggestion
    EXPECT_FALSE(value);
```

File: src/plugins/intel_cpu/src/exec_network.cpp (hunk 3 above)
Comment (vurusovs) [reply src/plugins/intel_cpu/src/exec_network.cpp:117]: Yes :-)

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp (hunk 4 above)
Comment (vurusovs) [reply src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:688]: Thanks, done!

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp [hunk 5]
```diff
@@ -679,6 +679,18 @@ TEST(OVClassBasicTest, SetTBBForceTerminatePropertyCoreNoThrow) {
...
+    OV_ASSERT_NO_THROW(ie.set_property(ov::enable_mmap(true)));
+    OV_ASSERT_NO_THROW(value = ie.get_property(ov::enable_mmap.name()).as<bool>());
+    EXPECT_EQ(value, true);
```
[the remark this answers is not in the document]
Comment (vurusovs) [reply src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:691]: Thanks, done!

File: src/frontends/ir/tests/frontend_test_basic.cpp (hunk 2 above)
Comment (vurusovs) [reply src/frontends/ir/tests/frontend_test_basic.cpp:358]: Thanks, done!

**Review by ilya-lavrenov — APPROVED** [review]

File: src/inference/src/ie_network_reader.hpp [hunk 6]
```diff
@@ -22,13 +22,15 @@ namespace details {
  * @param exts vector with extensions
  * @param ov_exts vector with OpenVINO extensions
  * @param newAPI Whether this function is called from OpenVINO 2.0 API
+ * @param ir_fe_use_map_allocator boolean to enable/disable MapAllocator in IR Frontend
```
Comment (vurusovs) [inline src/inference/src/ie_network_reader.hpp:25]: rename param and rewrite docstring

Comment (vurusovs) [reply src/inference/src/ie_network_reader.hpp]: Done

**Review by ilyachur — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**